### ISSUE TITLE
Carthage Warning: macro name is a reserved identifier [-Wreserved-id-macro]

### ISSUE THREAD

**Issue by mhillebrand** [PureLayout/PureLayout#162]
When including PureLayout via Carthage, I get these warnings:

```
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/app/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
```

**Comment by Lutzifer** [comment]
this should be fixed by #153, can you please check?

**Comment by mhillebrand** [comment]
@Lutzifer I'm new to Carthage. How would I specify [#153](https://github.com/PureLayout/PureLayout/pull/153) in my Cartfile?

**Comment by Lutzifer** [comment]
@mhillebrand 

That was confusing, indeed.

The correct Cartfile line is

`github "conichiGMBH/PureLayout" "antondomashnev/fix_macro_reserved_warning"`

You find a lot of examples for the Cartfile syntax on https://github.com/Carthage/Carthage/blob/master/Documentation/Artifacts.md#cartfile

**Comment by mhillebrand** [comment]
Thank you for the syntax wisdom, but that branch didn't alleviate the warnings:

```/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
/Users/mhillebrand/work/NewsApp/SinclairMobileSDK/Carthage/Checkouts/PureLayout/PureLayout/PureLayout/PureLayout+Internal.h:32:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
warning: section "__textcoal_nt" is deprecated
warning: section "__const_coal" is deprecated
warning: section "__textcoal_nt" is deprecated
warning: section "__const_coal" is deprecated
warning: section "__textcoal_nt" is deprecated
warning: section "__const_coal" is deprecated
warning: section "__textcoal_nt" is deprecated
warning: section "__const_coal" is deprecated
warning: section "__textcoal_nt" is deprecated
warning: section "__const_coal" is deprecated
warning: section "__textcoal_nt" is deprecated
warning: section "__const_coal" is deprecated
warning: section "__textcoal_nt" is deprecated
warning: section "__const_coal" is deprecated
warning: section "__textcoal_nt" is deprecated
warning: section "__const_coal" is deprecated```

**Comment by mhillebrand** [comment]
Also, when locking down the version number in my Cartfile, is it okay to leave off the initial letter "v" (e.g. `3.0.2` instead of `v3.0.2`)? I noticed that PureLayout is the only Carthage dependency I'm using that has the letter "v" at the beginning of its tag names, and Carthage barfs if I include it in my Cartfile:

`Parse error: unexpected trailing characters in line: github "PureLayout/PureLayout" == v3.0.2`

So I'm gonna replace that with:

`github "PureLayout/PureLayout" == 3.0.2`

...and cross my fingers.

**Comment by Lutzifer** [comment]
running `carthage update` using a cartfile with only the line

`github "conichiGMBH/PureLayout" "antondomashnev/fix_macro_reserved_warning"`

gives me:

```
*** Fetching PureLayout
*** Checking out PureLayout at "e40123d02f39ff1a768e539b56bf8455a1fe654f"
*** xcodebuild output can be found in /var/folders/vg/gf7092154qs10_l0rh6cfwym0000gn/T/carthage-xcodebuild.jQRdTo.log
*** Building scheme "PureLayout_Mac" in PureLayout.xcodeproj
*** Building scheme "PureLayout_tvOS" in PureLayout.xcodeproj
xcodebuild: error: Unable to find a destination matching the provided destination specifier:
		{ platform:tvOS Simulator, id:28D97318-3496-4ACA-82C2-AE06834AEB93 }

	Ineligible destinations for the "PureLayout_tvOS" scheme:
		{ platform:tvOS, id:dvtdevice-DVTiOSDevicePlaceholder-appletvos:placeholder, name:Generic tvOS Device }
		{ platform:tvOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-appletvsimulator:placeholder, name:Generic tvOS Simulator Device }
A shell task (/usr/bin/xcrun xcodebuild -project /private/var/folders/vg/gf7092154qs10_l0rh6cfwym0000gn/T/tmp.2wXV6ZHd/Carthage/Checkouts/PureLayout/PureLayout/PureLayout.xcodeproj -scheme PureLayout_tvOS -configuration Release -sdk appletvsimulator -destination "platform=tvOS Simulator,id=28D97318-3496-4ACA-82C2-AE06834AEB93" -destination-timeout 3 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 70:
xcodebuild: error: Unable to find a destination matching the provided destination specifier:
		{ platform:tvOS Simulator, id:28D97318-3496-4ACA-82C2-AE06834AEB93 }

	Ineligible destinations for the "PureLayout_tvOS" scheme:
		{ platform:tvOS, id:dvtdevice-DVTiOSDevicePlaceholder-appletvos:placeholder, name:Generic tvOS Device }
		{ platform:tvOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-appletvsimulator:placeholder, name:Generic tvOS Simulator Device }
```

Try removing your Carthage folder or running your carthage update in another temporary folder for testing.

**Comment by mhillebrand** [comment]
Boom. This looks good!

```
$ carthage update --platform iOS
*** Fetching PureLayout
*** Checking out PureLayout at "e40123d02f39ff1a768e539b56bf8455a1fe654f"
*** xcodebuild output can be found in /var/folders/ml/d9hzjcrd3mv_3hj_3qqf__brkdn63y/T/carthage-xcodebuild.cUymlw.log
*** Building scheme "PureLayout_iOS" in PureLayout.xcodeproj
```

**Comment by mhillebrand** [comment]
The problem is that my app and its SDK, which both rely on Carthage, are now fetching and building *two* versions of PureLayout—one from _PureLayout/PureLayout_ and one from _conichiGMBH/PureLayout_. I tried several times to correct this, but I'm stumped. I deleted the Carthage directory in both my SDK project and my app project, I deleted both Cartfile.resolved files, I deleted all my derived data, I deleted everything in _~/Library/Caches/org.carthage.CarthageKit_, and I emptied my trash.

How on Earth does Carthage have both PureLayout repos on its mind? I only have one listed (the same one) in both of my Cartfile files.

Thanks,
Matt
